src/components/reusable/checkbox/checkboxGroup.ts

```diff
@@ -83,6 +97,11 @@ export class CheckboxGroup extends LitElement {
   }
 
   override updated(changedProps: any) {
+    //check if any (internal / external )error msg. present then isInvalid is true
+    this.isInvalid =
+      this.invalidText !== '' || this.internalValidationMsg !== ''
+        ? true
+        : false;
```

Comment:
here, we are setting the value of **isInvalid**. So what i did is inside **update** function I am checking if there any validation (internal or external) is present then I m setting **isInvalid** is true, otherwise false

Reply:
Right, but anything within `updated()` should be wrapped in a check for `changedProps`, so it doesn't run unnecessarily on every update.

Reply:
you mean like this inside updated()

` if(this.invalidText !== '' || this.internalValidationMsg !== ''){
      this.isInvalid = true;
    } else {
      this.isInvalid = false;
    }`

Reply:
or 
if (changedProps('invalidText') || changedProps('internalValidationMsg')){
         this.isInvalid = this.invalidText !== '' || this.internalValidationMsg !== '' ? true : false;
}

Reply:
Done